src/modules/Elsa.MongoDB/HostedServices/CreateIndices.cs

```diff
@@ -0,0 +1,277 @@
+
+namespace Elsa.MongoDB.HostedServices;
+
+internal class CreateIndices : IHostedService
+{
+    private readonly IMongoCollection<Application> _applicationCollection;
+    private readonly IMongoCollection<User> _userCollection;
+    private readonly IMongoCollection<Role> _roleCollection;
+    private readonly IMongoCollection<WorkflowDefinitionLabel> _workflowDefinitionLabelCollection;
+    private readonly IMongoCollection<WorkflowDefinition> _workflowDefinitionCollection;
+    private readonly IMongoCollection<WorkflowInstance> _workflowInstanceCollection;
+    private readonly IMongoCollection<WorkflowState> _workflowStateCollection;
+    private readonly IMongoCollection<WorkflowExecutionLogRecord> _workflowExecutionLogCollection;
+    private readonly IMongoCollection<StoredBookmark> _workflowBookmarkCollection;
+    private readonly IMongoCollection<StoredTrigger> _workflowTriggerCollection;
+
+    public CreateIndices(
+        IMongoCollection<Application> applicationCollection,
+        IMongoCollection<User> userCollection,
+        IMongoCollection<Role> roleCollection,
+        IMongoCollection<WorkflowDefinitionLabel> workflowDefinitionLabelCollection,
+        IMongoCollection<WorkflowDefinition> workflowDefinitionCollection,
+        IMongoCollection<WorkflowInstance> workflowInstanceCollection,
+        IMongoCollection<WorkflowState> workflowStateCollection,
+        IMongoCollection<WorkflowExecutionLogRecord> workflowExecutionLogCollection,
+        IMongoCollection<StoredBookmark> workflowBookmarkCollection,
+        IMongoCollection<StoredTrigger> workflowTriggerCollection)
+    {
+        _applicationCollection = applicationCollection;
+        _userCollection = userCollection;
+        _roleCollection = roleCollection;
+        _workflowDefinitionLabelCollection = workflowDefinitionLabelCollection;
+        _workflowDefinitionCollection = workflowDefinitionCollection;
+        _workflowInstanceCollection = workflowInstanceCollection;
+        _workflowStateCollection = workflowStateCollection;
+        _workflowExecutionLogCollection = workflowExecutionLogCollection;
+        _workflowBookmarkCollection = workflowBookmarkCollection;
+        _workflowTriggerCollection = workflowTriggerCollection;
+    }
+
+    public Task StartAsync(CancellationToken cancellationToken)
+    {
+        return Task.WhenAll(
+            CreateApplicationIndices(cancellationToken),
+            CreateUserIndices(cancellationToken),
+            CreateRoleIndices(cancellationToken),
+            CreateWorkflowDefinitionLabelIndices(cancellationToken),
+            CreateWorkflowDefinitionIndices(cancellationToken),
+            CreateWorkflowInstanceIndices(cancellationToken),
+            CreateWorkflowStateIndices(cancellationToken),
+            CreateWorkflowExecutionLogIndices(cancellationToken),
+            CreateWorkflowBookmarkIndices(cancellationToken),
+            CreateWorkflowTriggerIndices(cancellationToken));
+    }
```

Comment:
Can we split this hosted service into ones that belong to their own (logical) module (like what is done with the EF Core provider)? 

Right now, even if my application doesn't use the Labels module, for example, my MongoDB database will have collections + indices related to Labels.